apps/full-site-editing/full-site-editing-plugin/full-site-editing/blocks/site-title/edit.js

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import React, { Component } from 'react';
+import { __ } from '@wordpress/i18n';
+const { withSelect } = wp.data;
+const { apiFetch } = wp;
+
+/**
+ * Internal dependencies
+ */
+
+class SiteTitleEdit extends Component {
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			title: null,
+			initialTitle: null,
+		};
+	}
+
+	componentDidMount() {
+		if ( this.state.initialTitle ) {
+			return;
+		}
+
+		apiFetch( { path: '/wp/v2/settings' } ).then( ( { title } ) => {
+			this.setState( { initialTitle: title } );
+		} );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { isSaving, isPublished, isPublishing, isAutosaving } = this.props;
+
+		if ( this.isPure() || prevProps.isSaving === this.props.isSaving ) {
```

Comment:
There's not much happening in `isPure`, so fine to put that inline. As a side-note, aim for naming where we don't need to look at code to figure out the intent. For example `titleHasNoChanges`